data-pipeline-flink/user-cache-updater-2.0/src/main/scala/org/sunbird/dp/usercache/util/UserMetadataUpdater.scala

```diff
@@ -46,265 +49,64 @@ object UserMetadataUpdater {
   }
 
   def getRegisteredUserInfo(userId: String, event: Event, metrics: Metrics, config: UserCacheUpdaterConfigV2, dataCache: DataCache,
-                            cassandraConnect: CassandraUtil, custodianRootOrgId: String): mutable.Map[String, AnyRef] = {
-    val userDetails: mutable.Map[String, AnyRef] = getUserDetails(config, cassandraConnect, metrics, userId, custodianRootOrgId)
-    val reportInfoMap: mutable.Map[String, AnyRef] =
-      if (!userDetails.isEmpty)
-        getReportInfo(userDetails, config, cassandraConnect, userId, metrics)
-      else {
-        logger.info(s"User DB does not have details for user: ${userId}")
-        metrics.incCounter(config.skipCount)
-        mutable.Map[String, AnyRef]()
+                            headers: Map[String, String], restUtil: RestUtil): mutable.Map[String, AnyRef] = {
+    var userCacheData: mutable.Map[String, AnyRef] = mutable.Map[String, AnyRef]()
+
+    val result = gson.fromJson[UserReadResult](restUtil.get(String.format("%s%s",config.userReadApiUrl, userId), Some(headers)), classOf[UserReadResult]).result
```

Comment:
This assumes that the API is always successful? what if the API returns an error code? how is it handled? Exception handling is missing

Reply:
If the status code is other than 200 it will not retry and the job will fail.